fairlearn/metrics/_metric_frame.py

```diff
@@ -117,24 +129,156 @@ def __init__(self,
                  y_pred, *,
                  sensitive_features,
                  control_features: Optional = None,
-                 sample_params: Optional[Union[Dict[str, Any], Dict[str, Dict[str, Any]]]] = None):
+                 sample_params: Optional[Union[Dict[str, Any], Dict[str, Dict[str, Any]]]] = None,
+                 streaming: bool = False):
         """Read a placeholder comment."""
-        check_consistent_length(y_true, y_pred)
+        if any(arg is None or len(arg) == 0 for arg in
+               (y_true, y_pred, sensitive_features)) and not streaming:
+            raise ValueError(_BAD_INIT_ERROR)
+
+        self._s_f = sensitive_features
+        self._c_f = control_features
+        self._s_p = sample_params
+        self._streaming = streaming
+        self.metric = metric
+
+        self._cf_names = None
+        self._sf_names = None
+
+        if not streaming:
+            # Since we are not streaming, we can compute the metrics already.
+            # This keeps the existing behavior where we compute the metrics in the
+            # constructor.
+            self._y_true = None
+            self._y_pred = None
+            check_consistent_length(y_true, y_pred)
+            y_t = _convert_to_ndarray_and_squeeze(y_true)
+            y_p = _convert_to_ndarray_and_squeeze(y_pred)
+            self._compute_metric(y_t, y_p, sensitive_features, control_features, sample_params)
+        else:
+            # If we are streaming, we wait, but initialize accumulators.
+            assert y_true == [] and y_pred == [] and sensitive_features == [], _NON_EMPTY_INIT_ERR
```

Comment:
This will eventually need to get changed to a ValueError, but that can wait until we're on to the polishing stage.

Reply:
done